Automattic/woocommerce-payments
# Refactoring Direct DB query in order_id_from_meta_key_value()

Partially Fixes #4151

#### Changes proposed in this Pull Request

Replaced the direct DB query in `order_id_from_meta_key_value` with `wc_get_orders`


#### Testing instructions

The method `order_id_from_meta_key_value` is used by `order_from_charge_id` and `order_from_intent_id` methods, the latter is used in `get_order_from_event_body_intent_id`, which is used in webhook processing. So we may test this change by processing a webhook.

1) Set Payment Settings to Capture Later so that the Order goes on hold and it is easy to observe the effect of `payment_intent.succeeded` webhook.
2) Place an order
3) POST a payment_intent.succeeded webhook, sample JSON
```
{
    "type": "payment_intent.succeeded",
    "data": {
        "object": {
            "id": "pi_INTENT_ID",
            "currency": "USD",
            "status": "succeeded",
            "charges": {
                "data": [
                    {
                        "id": "ch_CHARGE_ID"
                    }
                ]
            }
        }
    }
}
```
4) Check that the order moves to Processing and order notes are updated because the correct order is picked from the intent id.
5) The above steps should be tested under all the following conditions:
    -  No COT
    -  COT, using orders table and no sync
    -  COT, using posts table and no sync
    -  COT, using order table with sync
    -  COT, using posts table with sync

-------------------


**Post merge**
